src/dlsnode/storage/iterator/NeoStorageEngineStepIterator.d

```diff
@@ -251,20 +253,38 @@ public class NeoStorageEngineStepIterator
             switch (this.state)
             {
                 case State.Initializing:
-                    if (!this.openFirstBucket())
+                    bool wait_for_open;
+                    auto has_more_buckets = this.openFirstBucket(job_notification,
+                            wait_for_open);
+
+                    if (wait_for_open)
                     {
-                        return true;
+                        wait_for_data = true;
+                        return false;
```

Comment:
This return value implies that the documentation on this method's return value is outdated:
>true if there's no more data in the requested range, false otherwise

I mean, `wait_for_open` being true doesn't necessarily mean there's no more data, right?